src/coreclr/jit/morph.cpp

```diff
@@ -9473,7 +9474,7 @@ GenTree* Compiler::fgMorphConst(GenTree* tree)
     // guarantee slow performance for that block. Instead cache the return value
     // of CORINFO_HELP_STRCNS and go to cache first giving reasonable perf.
 
-    if (compCurBB->bbJumpKind == BBJ_THROW)
+    if ((compCurBB->bbJumpKind == BBJ_THROW) || compCurBB->isRunRarely())
     {
```

Comment:
Experimenting.. 
currently when we pass string literals to ThrowHelper-like calls they're materialized but should prefer LazyStringHelper instead.

Reply:
Alternative fix: handle args explicitly https://github.com/dotnet/runtime/pull/48589/commits/c6784d7e2ccf220ecc5a64f225e4eb8c6960e7bf